### DISCUSSION TITLE
Failed waiting for web/db containers to become ready: web container failed

### DISCUSSION THREAD

**Discussion by vardumper** [craftcms/image#9] [General]
A colleague of mine reported this issue (he tested with the image a few days back), probably before you switched back to ubuntu.

I had a similar issue with another DDEV install. The logs showed `Unable to create the PID file (/run/php/php8.3-fpm.pid).: Permission denied (13)` after I had added a custom Dockerfile. It turned out that both `/var/run/php-fpm.sock` as well as `/var/run/php` had gotten my host users permission. It doesn't make much sense, as the package I installed was simply the postgres-client v16.
The solution that worked, was to add a `RUN chmod 0777 /run/php` at the end of the Dockerfile (or at least after the apt commands).

Hope this helps.

A question what's PHP-SOAP doing in the repo? Does Craft really use it? 
A suggestion: could we have separate images for mysql / postgres? I'm not sure but I guess they would be more compact like that.

**Comment by jasonmccallister** [comment]
Hi @vardumper, thank you for the feedback. Are you by chance running Linux as your host?

Also, SOAP is a requirement for[ Craft Commerce](https://craftcms.com/docs/commerce/5.x/requirements.html) which is why its included in these images.

Also, happy to discuss including MySQL or PostgreSQL tools in the images, however the majority of Craft projects opt for MySQL over PostgreSQL, so I think it might be worth installing the MySQL tools by default.
